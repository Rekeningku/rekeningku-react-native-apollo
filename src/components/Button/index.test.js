import { Text } from 'react-native'
import React from 'react'
import sinon from 'sinon'
import { shallow } from 'enzyme'
import Button from './index'


describe('Button Component', () => {
  it('renders with all props', () => {
    const wrapper = shallow(
      <Button>
        First Text
      </Button>,
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('calls handler on press', () => {
    const onPressHandler = sinon.fake()
    const wrapper = shallow(
      <Button onPress={onPressHandler}>
        <Text>
          First Text
        </Text>
      </Button>,
    )
    wrapper.simulate('press')
    expect(onPressHandler.callCount).toBe(1)
  })
})
