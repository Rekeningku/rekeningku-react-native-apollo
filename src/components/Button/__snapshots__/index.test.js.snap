// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component renders with all props 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ButtonComponent>
    First Text
  </ButtonComponent>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Styled(Component)>
        First Text
      </Styled(Component)>,
      "onPress": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": "First Text",
      },
      "ref": null,
      "rendered": "First Text",
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Styled(Component)>
          First Text
        </Styled(Component)>,
        "onPress": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "First Text",
        },
        "ref": null,
        "rendered": "First Text",
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
